docs/cmdline-opts/write-out.d

```diff
@@ -50,6 +50,9 @@ curl CONNECT request. (Added in 7.12.4)
 .B http_version
 The http version that was effectively used. (Added in 7.50.0)
 .TP
+.B json
+A JSON object with all available keys.
```

Comment:
This could need some clarification. Does the `%{json}` string need to be placed anywhere in particular in the string and if so, what happens if not?

Reply:
`%{json}` is no different than all the others. if you leave `%{http_version}` out you wont receive it on stdout.

Reply:
Oh... you know, I somehow didn't consider that **with all available keys** means you'll dump them all unconditionally. I was thinking using the json tag would mean that you'd enable json mode and then you'd specify the variables you need output following that tag. Wouldn't that be a more suitable approach that is also more similar to how the option works already?

Reply:
honestly: no. the `%{json}` part is considered to be used programatically … i do not see any value in specifying a subset which imho is the niche and not the regular use case: hand over the fine metrics you have there in curl to something that is evaluating them.

to have only a subset on the command line because some innocent eyes are looking at the values: thats ok for me.

Reply:
i considered a dedicated --json mode which mimics the behaviour of --write-out … but then i was thinking about having to add a new --flag and a lot of documentation and some more in that regards. but then i realized that i can achieve the same effect with the existing command line

Reply:
Ok, I buy that reasoning.

But then why "overload" this into the `--writeout` option when it seems highly unlikely anyone would like to mix other output with `%{json}` ? Or can you actually think of a use case when someone would add some of the other variables as well?

Reply:
i use it with `%{stderr}` …

Reply:
so, how to progress? 

overloading semantics a bit but less hassle with a dedicated command OR new flag documention etc?

keeping `%{json}` in --write-out i can redirect it to stderr, i can prepend it by special markers, i can put "," between multiple lines and thus creating valid multi-entry-json ...

Reply:
Ok, let's do it like this. We always have the option to make a dedicated separate option later on, should we find more/better reasons to do so.

Reply:
i ll pull in your unit-test-branch